server/src/main/java/org/opensearch/search/backpressure/trackers/TaskResourceUsageTracker.java

```diff
@@ -26,14 +26,23 @@ public abstract class TaskResourceUsageTracker {
     /**
      * Counts the number of cancellations made due to this tracker.
      */
-    private final AtomicLong cancellations = new AtomicLong();
+    private final AtomicLong searchTaskCancellationCount = new AtomicLong();
+    private final AtomicLong searchShardTaskCancellationCount = new AtomicLong();
```

Comment:
We should not define search/shard task concepts on the TaskResourceUsageTracker abstraction level. This would otherwise mean that each TRUT requires to understand both search/shard tasks at the same time, which may not be true.

For eg. we may have a tracker that only cancels search tasks (eg. heavy aggregate queries).

Reply:
I still feel that the existing structure of the TaskResourceUsageTracker can remain intact. Instead of adding search/shard task concepts inside the tracker, we should have separate instances of a regular tracker for each task type.

Visually, it would look like this:
```
searchTaskTrackers = List.of(A, B)
searchShardTaskTrackers = List.of(B, C, D)
```

* A, B, C, D extend TaskResourceUsageTracker
* Search/shard tasks can now have different trackers, or even same trackers with different thresholds. The only important thing is that they are separate instances.
* No need to create separate methods for node stats.

Reply:
* Removed TaskType concept from the TRUT. 
* Separated out trackers for SearchTasks and SearchShardTasks as suggested by you.
* Removed separate methods for node stats.